docs/content/preview/architecture/transactions/read-restart-error.md

```diff
@@ -15,33 +15,71 @@ rightNav:
-* For writes with a write timestamp later than `T1` + `max_clock_skew`, the database can be sure that these writes were done after the read timestamp had been chosen. But for writes with a write timestamp between `T1` and `T1` + `max_clock_skew`, node `N2` can find itself in an ambiguous situation, such as the following:
+  Moreover, TSERVER 3's clock is running 5 units of time ahead of TSERVER 2's clock because of clock skew.
 
-  * It should return the data even if the client issued the read from a different node after writing the data, because the following guarantee needs to be maintained: the database should always return data that was committed in the past (past refers to the user-perceived past, and not based on machine clocks).
+1. An INSERT of token 29 is issued to YSQL on TSERVER 1.
+2. This INSERT is routed to the tablet hosted on TSERVER 3. The operation picks a commit time of `T2=103` even though the "true time" is `98`.
+3. TSERVER 1 acknowledges the INSERT.
+4. Now that the INSERT command is complete, the SELECT command is issued.
+5. TSERVER 1 starts a distributed read. TSERVER 1 reads data from multiple shards on different physical YB-TServers in the cluster, namely TSERVER 2 and TSERVER 3. The read point that defines the snapshot of the database at which the data will be read, is picked on TSERVER 2 based on the safe time of that YB-TServer, namely `T1=101`.
+6. TSERVER 1 gets back token 17 in the result set.
+7. TSERVER 1 now issues a read operation to TSERVER 3 to retrieve records from the second shard. However, since the read time `T1=101` is less than the commit time of the prior insert `T2=103`, the record is not part of the read snapshot.
+8. Thus, token 29 is omitted from the result set returned.
 
-  * It should not return the data if the write is performed in the future (that is, after the read point had been chosen) **and** had a write timestamp later than the read point. Because if that is allowed, everything written in future would be trivially visible to the read. Note that the latter condition is important because it is okay to return data that was written after the read point was picked, if the write timestamp was earlier than the read point (this doesn't break any consistency or isolation guarantees).
+  More generally, this anomaly occurs whenever the commit time of a prior write operation `T2` is higher than the read time `T1` of a later read operation, thus violating (distributed) consistency.
 
-* If node `N2` finds writes in the range `(T1, T1+max_clock_skew]`, to avoid breaking the strong guarantee that a reader should always be able to read what was committed earlier, and to avoid reading data with a later write timestamp which was also actually written after the read had been issued, node `N2` raises a `Read restart` error.
+How does YugabyteDB mitigate this clock skew anomaly?
```

Comment:
Would move this part up to after "YugabyteDB has optimizations to resolve this ambiguity internally with best-effort." given this appears to be an explanation of YBDB optimizations

Reply:
This is not one of the optimizations. We never describe the optimizations, just that they are present. Perhaps, mitigate is not the correct word to use here? I changed it to prevent. This basically describes how we raise a read restart error whenever we detect the anomaly. I hope that makes it clearer.